gstreamer/src/pad.rs

```diff
@@ -311,7 +312,7 @@ impl<O: IsA<Pad>> PadExtManual for O {
         }
     }
 
-    fn query(&self, query: &mut QueryRef) -> bool {
+    fn query(&self, query: &mut Query) -> bool {
```

Comment:
Note that the same pattern does not work for events/messages/buffers/etc, as you pass those by value instead of borrowing them (mutably). That's the special part about queries here

Reply:
In that case, we could implement something like `try_to_owned` that would transfer ownership of the `xxView` when `is_writable` is `true`.